scripts/ci/libraries/_build_images.sh

```diff
@@ -394,30 +394,129 @@ function build_images::get_docker_image_names() {
-function build_image::login_to_github_registry_if_needed() {
+# If GitHub Registry is used, login to the registry using GITHUB_USERNAME and
+# either GITHUB_TOKEN or CONTAINER_REGISTRY_TOKEN depending on the registry.
+# In case Personal Access token is not set, skip logging in
+function build_image::login_to_github_container_registry_if_github_registry_enabled() {
     if [[ ${USE_GITHUB_REGISTRY} == "true" ]]; then
-        if [[ -n ${GITHUB_TOKEN=} ]]; then
-            start_end::group_start "Login to GitHub registry"
-            echo "${GITHUB_TOKEN}" | docker login \
+        start_end::group_start "Determine Github Registry token used"
+        local token=""
+        if [[ "${GITHUB_REGISTRY}" == "ghcr.io" ]]; then
+            # For now ghcr.io can only authenticate using Personal Access Token with package access scope.
+            # There are plans to implement GITHUB_TOKEN authentication but this is not implemented yet
+            token="${CONTAINER_REGISTRY_TOKEN=}"
+            echo
+            echo "Using CONTAINER_REGISTRY_TOKEN!"
+            echo
+        elif [[ "${GITHUB_REGISTRY}" == "docker.pkg.github.com" ]]; then
+            token="${GITHUB_TOKEN}"
+            echo
+            echo "Using GITHUB_TOKEN!"
+            echo
+        else
+            echo
+            echo  "${COLOR_RED}ERROR: Bad value of '${GITHUB_REGISTRY}'. Should be either 'ghcr.io' or 'docker.pkg.github.com'!${COLOR_RESET}"
+            echo
+            exit 1
+        fi
+        if [[ "${token}" == "" ]] ; then
+            echo
+            echo "Skip logging in to Github Registry. No Token available!"
+            echo
+        fi
+        start_end::group_end
+        if [[ "${token}" != "" ]]; then
+            start_end::group_start "Login to GitHub Registry ${GITHUB_REGISTRY}"
+            echo "${token}" | docker login \
                 --username "${GITHUB_USERNAME:-apache}" \
                 --password-stdin \
                 "${GITHUB_REGISTRY}"
             start_end::group_end
+        else
+            start_end::group_start "Skip Login to GitHub Registry ${GITHUB_REGISTRY} as token is missing"
+            start_end::group_end
         fi
     fi
 }
 
+function build_image::enable_docker_experimental_features_if_github_registry_enabled() {
+    if [[ ${USE_GITHUB_REGISTRY} == "true" ]]; then
+        start_end::group_start "Enable experimental features in Docker ('docker manifest' command)"
+        python <<EOF
+import json
+from os.path import expanduser, join
+
+DOCKER_CONFIG_FILE=join(expanduser("~"), ".docker", 'config.json')
+with open(DOCKER_CONFIG_FILE) as f:
+    config = json.load(f)
+print("\nBefore:\n")
+print(json.dumps(config, indent=4))
+print("\nAdding experimental flag to enable docker manifest command\n")
+config['experimental'] = "enabled"
+print("\nAfter:\n")
+print(json.dumps(config, indent=4))
+with open(DOCKER_CONFIG_FILE,"wt") as f:
+    json.dump(config, f)
+EOF
+        start_end::group_end
```

Comment:
This is possible with much less code

```suggestion
           local new_config
           new_config="$( jq '.experimental = "enabled"' ~/.docker/config.json)
           echo "new_config" > ~/.docker/config.json
```

Reply:
We just have to make sure jqis installed.  But yeah. AFAIK is installed for GitHub Runner. Just make sure it is also there for the self-hosted run :)

Reply:
Yeah, it's installed there too - some of the actions expect it, so we have to mirror the environment of hosted runners